src/mame/nmk/nmk16.cpp

```diff
@@ -5118,6 +5148,34 @@ void nmk16_state::decode_gfx()
 	}
 }
 
+void tdragon_prot_state::decode_sabotenb()
+{
+	u8 *rom;
+	int len;
+
+	// background tiles
+	rom = memregion("bgtile")->base();
+	len = memregion("bgtile")->bytes();
+	for (int A = 0; A < len; A++)
+	{
+		rom[A] = m_nmk214[1]->decode_byte(A, rom[A]);
+	}
+
+	// Sprites
+	rom = memregion("sprites")->base();
+	len = memregion("sprites")->bytes();
+	for (int A = 0; A < len; A += 2)
+	{
+		// When the sprite ROM is loaded, it's also byte-swapped, so we keep the same order here to compound the word (A: second-byte, A+1: first-byte)
+		u16 word = (rom[A] << 8) | rom[A+1];
+		// As Sprites ROM works in word mode, address value here is in terms of bytes and should be divided by 2 to get the proper base-16 address value
+		u16 decoded_word = m_nmk214[0]->decode_word(A/2, word);
+		
+		rom[A] = decoded_word >> 8;
+		rom[A+1] = decoded_word & 0xff;
+	}
+}
```

Comment:
Is there a potential for issues with caching in the `gfxdecode_device`?  I’d expect you’d need to use `mark_dirty` in a loop to ensure tiles are refreshed.

Reply:
Now, I included tilemap function `mark_all_dirty()` for invalidate cached tiles, not sure if that's enough to refresh all the tiles or I need to do it in a loop as you mentioned. I didn't see any errors while testing.

Additionally, is this also needed for sprites?